test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,35 @@
+require 'test_helper'
+
+class UsersControllerTest < ActionController::TestCase
+
+  test "should get new" do
+    get :new
+    assert_response :success
+  end
+
+ test "should create user" do
+    assert_difference('User.count') do
+      post :create, user: { email: "melissa@melissa.com", name: "melissa", user_name: "supermeli", password: "davidpapacito" }
+    end
+
+    assert_redirected_to login_path
+  end
+
+  test "should not create user" do
```

Comment:
Names for tests should also express why they do not pass when you want them to not pass.
So something like "does not create user without a name" or "does not create a user with invalid attributes."